packages/sage/src/commands/codemod.ts

```diff
@@ -0,0 +1,105 @@
+import { Command } from 'commander'
+import { Dirent } from 'node:fs'
+import { access, readFile, readdir, rename, writeFile } from 'node:fs/promises'
+import path from 'node:path'
+import { cmd, exec, getPackageManager } from '../core/utils.js'
+import { logger } from '@roboplay/robo.js'
+import inquirer from 'inquirer'
+import { PackageJson } from 'src/core/types.js'
+
+const command = new Command('typescript')
+	.description('Turns your Javascript project into Typescript!')
+	.action(codemodAction)
+
+export default command
+
+interface ProjectInfo {
+	hasSWC: boolean
+}
+
+const TSCONFIG = `{
+	"compilerOptions": {
+		"target": "ESNext",
+		"lib": ["esnext"],
+		"allowJs": true,
+		"skipLibCheck": true,
+		"strict": true,
+		"forceConsistentCasingInFileNames": true,
+		"esModuleInterop": true,
+		"module": "esnext",
+		"moduleResolution": "node",
+		"resolveJsonModule": true,
+		"isolatedModules": true,
+		"incremental": true
+	},
+	"include": ["**/*.ts"],
+	"exclude": ["node_modules"]
+}`
+
+async function codemodAction() {
+	const packageManager = getPackageManager()
+	const tsconfigPath = path.join(process.cwd(), 'tsconfig.json')
+	const packageJsonPath = path.join(process.cwd(), 'package.json')
+	const packageJson: PackageJson = JSON.parse(await readFile(packageJsonPath, 'utf-8'))
+	const projectInfo: ProjectInfo = {
+		hasSWC: !!packageJson.devDependencies['@swc/core'] || !!packageJson.dependencies['@swc/core']
+	}
+
+	try {
+		if (!projectInfo.hasSWC) {
+			await exec(`${cmd(packageManager)} install @swc/core --save-dev`)
```

Comment:
Please move the `--save-dev` to be before the names of the packages. Some package managers won't recognize flags that come after positional arguments.

Also, please account for different variants. In PNPM and Yarn, it's called `add` rather than `install`. The reason we use `install` in other parts of the CLI is because we're installing the dependencies as a whole, rather than adding new ones.

If you need a good reference, the Pilot.js `create-pilot-app` CLI may show how to achieve this nicely.

Reply:
const command = packageManager === 'npm' ? 'install' : 'add'